JavaScript/JavaScriptSDK/appInsights.ts

```diff
@@ -301,7 +318,31 @@ module Microsoft.ApplicationInsights {
 
                 this.context.track(envelope);
             } catch (e) {
-                _InternalLogging.warnToConsole("trackTrace failed, trace will not be collected: " + Util.dump(e));
+                _InternalLogging.throwInternalNonUserActionable(LoggingSeverity.WARNING, "trackTrace failed, trace will not be collected: " + Util.dump(e));
+            }
+        }
+
+        /**
+        * Log a page visit time
+        * @param    pageName    Name of page
+        * @param    pageVisitDuration Duration of visit to the page in milleseconds
+        */
+        public trackPageVisitTime(pageName: string, pageUrl: string, pageVisitTime: number) {
```

Comment:
This is just for "phase 1" to unblock the development of this feature, until the page visit time standard metric is supported by the backed...